build.go

```diff
@@ -82,6 +110,34 @@ func Build(
 			return packit.BuildResult{}, fmt.Errorf("could not remove temp directory: %w", err)
 		}
 
+		if globalNugetPath != "" {
+			err = symlinker.Unlink(filepath.Join(homeDir, ".nuget", "NuGet", "NuGet.Config"))
+			if err != nil {
+				return packit.BuildResult{}, err
+			}
+		}
+
 		return packit.BuildResult{}, nil
 	}
 }
+
+func getBinding(typ, provider, bindingsRoot, entry string, bindingResolver BindingResolver, logger scribe.Logger) (string, error) {
+	bindings, err := bindingResolver.Resolve(typ, provider, bindingsRoot)
+	if err != nil {
+		return "", err
+	}
+
+	if len(bindings) > 1 {
+		return "", errors.New("binding resolver found more than one binding of type 'nugetconfig'")
+	}
+
+	if len(bindings) == 1 {
+		logger.Process("Loading nuget service binding")
+
+		if _, ok := bindings[0].Entries[entry]; !ok {
+			return "", fmt.Errorf("binding of type %s does not contain required entry %s", typ, entry)
+		}
+		return filepath.Join(bindings[0].Path, entry), nil
+	}
+	return "", nil
+}
```

Comment:
I think that all of this logic could be replace by [`bindingResolver.ResolveOne()`](https://github.com/paketo-buildpacks/packit/blob/f49887740eadf5f5117aaaeb7e4e1bbd52a2811a/servicebindings/resolver.go#L83-L92) the logic between the two looks identical unless I am missing something.

Reply:
`ResolveOne` fails [if the number of bindings != 1](https://github.com/paketo-buildpacks/packit/blob/f49887740eadf5f5117aaaeb7e4e1bbd52a2811a/servicebindings/resolver.go#L88). This means it fails when there are 0 bindings of type nuget. We want this binding to be optional, so we can't use `ResolveOne()`

Reply:
@ForestEckhardt `ResolveOne` actually doesn't quite work for our use case. The implementation relies on the idea that there will be exactly one binding tor resolve, and fails if there is not. In the case where there is no binding at all, it causes failures. I think it's better to leave this as is for now to get the functionality out and available, and then go and implement a breaking change to `ResolveOne` down the road to change the implementation.

Reply:
I see. Hmmm, that is not the functionality I would hope for from that function. I might have to give that a prod latter